src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -14591,5 +14636,423 @@ static int Test1(out int x)
+                // (8,23): error CS0270: Array size cannot be specified in a variable declaration (try initializing with a 'new' expression)
+                //         int c[out var x3] = null; // fatal syntax error - 'out' is skipped
+                Diagnostic(ErrorCode.ERR_ArraySizeInDeclaration, "x3").WithLocation(8, 23),
+                // (7,13): error CS0501: 'b(out var)' must declare a body because it is not marked abstract, extern, or partial
+                //         int b(out var x2) = null; // parsed as a local function with syntax error
+                Diagnostic(ErrorCode.ERR_ConcreteMissingBody, "b").WithArguments("b(out var)").WithLocation(7, 13),
+                // (7,19): error CS0825: The contextual keyword 'var' may only appear within a local variable declaration or in script code
+                //         int b(out var x2) = null; // parsed as a local function with syntax error
+                Diagnostic(ErrorCode.ERR_TypeVarNotFound, "var").WithLocation(7, 19),
+                // (8,29): error CS0037: Cannot convert null to 'int' because it is a non-nullable value type
+                //         int c[out var x3] = null; // fatal syntax error - 'out' is skipped
+                Diagnostic(ErrorCode.ERR_ValueCantBeNull, "null").WithArguments("int").WithLocation(8, 29),
+                // (8,19): error CS0103: The name 'var' does not exist in the current context
+                //         int c[out var x3] = null; // fatal syntax error - 'out' is skipped
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "var").WithArguments("var").WithLocation(8, 19),
+                // (8,23): error CS0103: The name 'x3' does not exist in the current context
+                //         int c[out var x3] = null; // fatal syntax error - 'out' is skipped
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "x3").WithArguments("x3").WithLocation(8, 23),
+                // (7,9): error CS0177: The out parameter 'x2' must be assigned to before control leaves the current method
+                //         int b(out var x2) = null; // parsed as a local function with syntax error
+                Diagnostic(ErrorCode.ERR_ParamUnassigned, "int b(out var x2) ").WithArguments("x2").WithLocation(7, 9),
+                // (6,25): warning CS0219: The variable 'a' is assigned but its value is never used
+                //         int[out var x1] a = null; // fatal syntax error - 'out' is skipped
+                Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "a").WithArguments("a").WithLocation(6, 25),
+                // (7,13): warning CS0168: The variable 'b' is declared but never used
+                //         int b(out var x2) = null; // parsed as a local function with syntax error
+                Diagnostic(ErrorCode.WRN_UnreferencedVar, "b").WithArguments("b").WithLocation(7, 13)
+                );
+        }
+
+        [Fact]
+        public void ElementAccess_06()
```

Comment:
`OutVariableDeclarationInIndex` is a success case. There is no success case for indexing an array.

---
In reply to: [75923726](https://github.com/dotnet/roslyn/pull/13274#discussion_r75923726) [](ancestors = 75923726)